### ISSUE TITLE
Issue with SCAFR performance in example code

### ISSUE THREAD

**Issue by Yoontae6719** [clabrugere/pytorch-scarf#3]
I wanted to bring to your attention a problem I discovered while working with the example code. It appears that the SCAFR result proposed in the code has lower performance than a simple logistic model.

I would appreciate any insights or suggestions you might have on this issue. Thank you in advance for your help.

**Comment by MarkYangjiayi** [comment]
I think in the example code the author just did a linear probing after pretrained the embedding. Ideally if you what the best performance you should also fine-tune the encoder like the paper did.

**Comment by clabrugere** [comment]
Thanks @Yoontae6719, as @MarkYangjiayi mentioned, the example notebook just applies a linear model to the learned embeddings and uses the labels to check if the embedding model did manage to capture the structure in the data. 
If the embeddings did not carry useful information, we would expect the linear model to have performance as bad as a random model. 
While the performance here is not as good as using the supervised dataset directly, it still manage to provide a good prediction performance. In a real setting, you would not have all the labels (or not at all in a fully unsupervised setting) so you could hardly expect to have the best performance anyway, you would train longer and you would also fine-tune it.

**Comment by MarkYangjiayi** [comment]
Hi @clabrugere , do you mind sharing your experience in using this pre-training approach in industry data? As the original author didn't provide official code, I wasn't really confident in trying out this method in my projects. Did you find any improvements (if any) claimed in the original paper using your implementation?

**Comment by clabrugere** [comment]
Hi @MarkYangjiayi to be fair I never used this model on real data so I can't really say if it works well on real noisy data sorry.. Maybe you could try to replicate it on a dataset you know and which has label to see if it generates embeddings that make sense for your use case.
